Implement Jacobi amplitude

pleroy opened the pull request:
```
Run on (4 X 3310 MHz CPU s)
CPU Caches:
  L1 Data 32K (x4)
  L1 Instruction 32K (x4)
  L2 Unified 262K (x4)
  L3 Unified 6291K (x1)
-----------------------------------------------------------------
Benchmark                          Time           CPU Iterations
-----------------------------------------------------------------
BM_JacobiAmplitude               186 ns        184 ns   15090000
BM_JacobiAmplitude               194 ns        189 ns   15090000
BM_JacobiAmplitude               185 ns        186 ns   15090000
BM_JacobiAmplitude               185 ns        184 ns   15090000
BM_JacobiAmplitude               186 ns        184 ns   15090000
BM_JacobiAmplitude               186 ns        186 ns   15090000
BM_JacobiAmplitude               184 ns        185 ns   15090000
BM_JacobiAmplitude               185 ns        183 ns   15090000
BM_JacobiAmplitude               185 ns        183 ns   15090000
BM_JacobiAmplitude               184 ns        185 ns   15090000
BM_JacobiAmplitude_mean          186 ns        185 ns         10
BM_JacobiAmplitude_median        185 ns        185 ns         10
BM_JacobiAmplitude_stddev          3 ns          2 ns         10
BM_JacobiSNCNDN                  131 ns        131 ns   21890000
BM_JacobiSNCNDN                  133 ns        131 ns   21890000
BM_JacobiSNCNDN                  132 ns        132 ns   21890000
BM_JacobiSNCNDN                  132 ns        132 ns   21890000
BM_JacobiSNCNDN                  132 ns        130 ns   21890000
BM_JacobiSNCNDN                  132 ns        131 ns   21890000
BM_JacobiSNCNDN                  131 ns        132 ns   21890000
BM_JacobiSNCNDN                  131 ns        130 ns   21890000
BM_JacobiSNCNDN                  131 ns        131 ns   21890000
BM_JacobiSNCNDN                  134 ns        135 ns   21890000
BM_JacobiSNCNDN_mean             132 ns        131 ns         10
BM_JacobiSNCNDN_median           132 ns        131 ns         10
BM_JacobiSNCNDN_stddev             1 ns          1 ns         10
```

eggrobin commented on numerics/elliptic_functions.cpp:
```diff
@@ -197,6 +199,65 @@ void JacobiSNCNDN(double const u,
+  } else {
+    double const k = EllipticK(mc);
+    JacobiSNCNDNWithK(u, mc, k, s, c, d);
+    offset = 2.0 * π * std::round(u / (4.0 * k)) * Radian;
```
why not `std::nearbyint`?

pleroy replied:
Done.

pleroy commented:
PTAL, I added a monotonicity test and fixed the implementation.  New benchmark:
```
Run on (4 X 3310 MHz CPU s)
CPU Caches:
  L1 Data 32K (x4)
  L1 Instruction 32K (x4)
  L2 Unified 262K (x4)
  L3 Unified 6291K (x1)
-----------------------------------------------------------------
Benchmark                          Time           CPU Iterations
-----------------------------------------------------------------
BM_JacobiAmplitude               199 ns        200 ns   14360000
BM_JacobiAmplitude               200 ns        199 ns   14360000
BM_JacobiAmplitude               200 ns        198 ns   14360000
BM_JacobiAmplitude               199 ns        199 ns   14360000
BM_JacobiAmplitude               199 ns        198 ns   14360000
BM_JacobiAmplitude               199 ns        198 ns   14360000
BM_JacobiAmplitude               199 ns        199 ns   14360000
BM_JacobiAmplitude               198 ns        196 ns   14360000
BM_JacobiAmplitude               199 ns        197 ns   14360000
BM_JacobiAmplitude               199 ns        200 ns   14360000
BM_JacobiAmplitude_mean          199 ns        198 ns         10
BM_JacobiAmplitude_median        199 ns        198 ns         10
BM_JacobiAmplitude_stddev          1 ns          1 ns         10
```

pleroy commented:
retest this please
